### PR TITLE
Upgrade embulk-base-restclient to v0.5.3 and fixed hang job

### PR DESCRIPTION
**Pull request by thangnc**
- Upgraded `embulk-base-restclient` to v0.5.3 to fix hang job

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,16 +18,16 @@ configurations {
...
 sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 dependencies {
-    compile  "org.embulk:embulk-core:0.8.15"
-    provided "org.embulk:embulk-core:0.8.15"
-    compile  "org.embulk.base.restclient:embulk-base-restclient:0.5.2"
-    compile  "org.embulk.base.restclient:embulk-util-retryhelper-jetty92:0.5.2"
+    compile  "org.embulk:embulk-core:0.8.24"
+    provided "org.embulk:embulk-core:0.8.24"
```
Comment (sakama) [inline build.gradle]: I'm not sure if using Embulk v0.8.24 is safer nor not.
It was released just yesterday.

**Review by sakama — COMMENTED** [review]
Put 1 minor comment.
Everthing else looks good to me.

File: build.gradle (hunk 1 above)
Comment (sakama) [reply build.gradle]: Are you going to take RetryHelper fix at another PR?

Comment (thangnc) [reply build.gradle]: Yes, im trying to close connection before throw `ConfigException`

**Comment by thangnc** [comment]
@sakama PTAL

**Review by muga — APPROVED** [review]
LGTM. To fix hang job issue, this PR is reasonable. 

This is minor comment: I feel that we don't have big reasons why we have 2 record buffer classes: `MailChimpAbstractRecordBuffer` and `MailChimpRecordBuffer`. Rather, the methods declared in this `MailChimpRecordBuffer` should be in `MailChimpHttpClient`.

**Comment by thangnc** [comment]
@muga actually, i want to keep `MailChimpHttpClient` like the wrapper of http client to sending and receiving, parsing data. The main logic will be in `RecordBuffer` classes. 

- The reason i created `MailChimpAbstractRecordBuffer` to i can do mock and test the business logic of `MailChimpRecordBuffer`

**Comment by muga** [comment]
@thangnc Agree. It's better to have the wrapper of raw http client. and I understand the motivation of `MailChimpHttpClient`. How about `MailChimpRecordBuffer`? It doesn't have buffer like functionalities. I suggest like
- Rename `MailChimpRecordBuffer` with `MailChimpClient`. Or merge the functionalities into `MailChimpHttpClient`.
- Rename `MailChimpAbstractRecordBuffer` with `MailChimpRecordBuffer`. and then, the class has a `MailChimpClient` object as the instance field. 

How about them?

**Comment by thangnc** [comment]
@muga ok. i will do a refactor like your suggestion in other PR

**Pull request merged**